libs/sdk-backend-tiger/src/convertors/fromBackend/afm/result.ts

```diff
@@ -131,3 +133,18 @@ export function transformExecutionResult(
         total: result.paging.total,
     };
 }
+
+export function transformGrandTotalData(
+    definition: IExecutionDefinition,
+    grandTotals: ResGrandTotal[],
+): DataValue[][][] {
+    const grandTotalsData: DataValue[][][] = definition.dimensions.map((_) => []);
+    const grandTotalsByLocalId = new Map(grandTotals.map((total) => [total.localIdentifier, total.data]));
+    withTotals(definition.dimensions, (dimIdx, typeIdx, totalsOfType) => {
+        const totalType = totalsOfType[0].type;
+        const localId = totalLocalIdentifier(totalType, dimIdx);
+        // Tiger API supports multi-dimensional totals but SDK limits to single dim totals only
+        grandTotalsData[dimIdx][typeIdx] = grandTotalsByLocalId.get(localId)!! as Array<DataValue>;
```

Comment:
i'm curious how does the grandTotals array look like in situation when table has two measures and the backend is asked to calculate for just one.. say the first one? what will be size of the array? 2 with the first element the calculated total and second null?

asking because this is not the documented aspect of the totals contract (i'll spend some time documenting this shortly while we are in the area)

Reply:
I intended returning the same data as current bear, i.e. `[ number, null ]`. Do you know of any better way? IMO we don't actually have any options...